src/MultiFormats/NetworkProtocol.cs

```diff
@@ -0,0 +1,580 @@
+    {
+        internal static Dictionary<string, Type> Names = new Dictionary<string, Type>();
+        internal static Dictionary<uint, Type> Codes = new Dictionary<uint, Type>();
+
+        /// <summary>
+        ///   Registers the standard network protocols for IPFS.
+        /// </summary>
+        static NetworkProtocol()
+        {
+            Register<Ipv4NetworkProtocol>();
+            Register<Ipv6NetworkProtocol>();
+            Register<TcpNetworkProtocol>();
+            Register<UdpNetworkProtocol>();
+            Register<P2PNetworkProtocol>();
+            RegisterAlias<IpfsNetworkProtocol>();
+            Register<QuicNetworkProtocol>();
+            Register<HttpNetworkProtocol>();
+            Register<HttpsNetworkProtocol>();
+            Register<DccpNetworkProtocol>();
+            Register<SctpNetworkProtocol>();
+            Register<WsNetworkProtocol>();
+            Register<Libp2PWebrtcStarNetworkProtocol>();
+            Register<UdtNetworkProtocol>();
+            Register<UtpNetworkProtocol>();
+            Register<OnionNetworkProtocol>();
+            Register<Libp2PWebrtcDirectNetworkProtocol>();
+            Register<P2PCircuitNetworkProtocol>();
+            Register<DnsNetworkProtocol>();
+            Register<Dns4NetworkProtocol>();
+            Register<Dns6NetworkProtocol>();
+            Register<DnsAddrNetworkProtocol>();
+            Register<WssNetworkProtocol>();
+            Register<IpcidrNetworkProtocol>();
+        }
+
+        /// <summary>
+        ///   Register a network protocol for use.
+        /// </summary>
+        /// <typeparam name="T">
+        ///   A <see cref="NetworkProtocol"/> to register.
+        /// </typeparam>
+        public static void Register<T>() where T : NetworkProtocol, new()
+        {
+            var protocol = new T();
+
+            if (Names.ContainsKey(protocol.Name))
+                throw new ArgumentException(string.Format("The IPFS network protocol '{0}' is already defined.",
+                    protocol.Name));
+            if (Codes.ContainsKey(protocol.Code))
+                throw new ArgumentException(string.Format("The IPFS network protocol code ({0}) is already defined.",
+                    protocol.Code));
+
+            Names.Add(protocol.Name, typeof(T));
+            Codes.Add(protocol.Code, typeof(T));
+        }
+
+        /// <summary>
+        ///   Register an alias to another network protocol.
+        /// </summary>
+        /// <typeparam name="T">
+        ///   A <see cref="NetworkProtocol"/> to register.
+        /// </typeparam>
+        public static void RegisterAlias<T>() where T : NetworkProtocol, new()
+        {
+            var protocol = new T();
+
+            if (Names.ContainsKey(protocol.Name))
+                throw new ArgumentException(string.Format("The IPFS network protocol '{0}' is already defined.",
+                    protocol.Name));
+            if (!Codes.ContainsKey(protocol.Code))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)